Sources/PointFree/TODO.swift

```diff
@@ -313,3 +300,122 @@ extension PartialIso where A == B.RawValue, B: RawRepresentable {
 public func mapExcept<E, F, A, B>(_ f: @escaping (Either<E, A>) -> Either<F, B>) -> (EitherIO<E, A>) -> EitherIO<F, B> {
   return { $0.mapExcept(f) }
 }
+
+public protocol TaggedType {
+  associatedtype _Tag
+  associatedtype _A
+
+  var unwrap: _A { get }
+  init(unwrap: _A)
+}
+
+extension Tagged: TaggedType {
+  public typealias _Tag = Tag
+  public typealias _A = A
+}
+
+extension PartialIso where A: Codable, B: TaggedType, A == B._A {
+  public static var tagged: PartialIso<B._A, B> {
+    return PartialIso(
+      apply: B.init(unwrap:),
+      unapply: ^\.unwrap
+    )
+  }
+}
+
+extension PartialIso where A == String, B == UUID {
+  public static var uuid: PartialIso<String, UUID> {
+    return PartialIso(
+      apply: UUID.init(uuidString:),
+      unapply: ^\.uuidString
+    )
+  }
+}
+
+public func lift<A>(_ a: A) -> Tuple1<A> {
+  return Tuple1(first: a, second: unit)
+}
+
+extension IO {
+  public var parallel: Parallel<A> {
+    return Parallel { callback in
+      callback(self.perform())
+    }
+  }
+}
+
+extension EitherIO {
+  func retry(maxRetries: Int) -> EitherIO {
+    return retry(maxRetries: maxRetries, backoff: const(.seconds(0)))
+  }
+
+  func retry(maxRetries: Int, backoff: @escaping (Int) -> DispatchTimeInterval) -> EitherIO {
+    return self.retry(maxRetries: maxRetries, attempts: 0, backoff: backoff)
+  }
+
+  private func retry(maxRetries: Int, attempts: Int, backoff: @escaping (Int) -> DispatchTimeInterval) -> EitherIO {
+
+    return self <|> .init(run:
+      self
+        .run
+        .delay(backoff(attempts))
+    )
+  }
+
+  public func delay(_ interval: DispatchTimeInterval) -> EitherIO {
+    return .init(run: self.run.delay(interval))
+  }
+}
+
+extension DispatchTimeInterval {
+  private var nanoseconds: Int? {
+    switch self {
+    case let .seconds(n):
+      return .some(n * 1_000_000_000)
+    case let .milliseconds(n):
+      return .some(n * 1_000_000)
+    case let .microseconds(n):
+      return .some(n * 1_000)
+    case let .nanoseconds(n):
+      return .some(n)
+    case .never:
+      return nil
+    }
+  }
+
+  public static func + (lhs: DispatchTimeInterval, rhs: DispatchTimeInterval) -> DispatchTimeInterval {
+    return (curry(+) <¢> lhs.nanoseconds <*> rhs.nanoseconds)
+      .map(DispatchTimeInterval.nanoseconds)
+      ?? .never
+  }
+}
+
+extension IO {
+  public func delay(_ interval: DispatchTimeInterval) -> IO {
+    return .init { callback in
+      DispatchQueue.global().asyncAfter(deadline: .now() + interval) {
+        callback(self.perform())
+      }
+    }
+  }
+}
+
+func zip<A>(_ parallels: [Parallel<A>]) -> Parallel<[A]> {
```

Comment:
I think this is a traversable, `mapM` kinda thing.

``` haskell
mapM :: (Monad m, Traversable t) => (a -> m b) -> t a -> m (t b)
```

Reply:
yeah prob, also my implementation here is prob not very thread safe? should I been locking around the `count` check?

Reply:
Yeah. Let's throw the check and increment into a queue.